src/Transport/Sending/MessageDispatcher.cs

```diff
@@ -120,6 +120,7 @@ public async Task Dispatch(TransportOperations outgoingMessages, TransportTransa
                 var operations = destinationAndOperations.Value;
 
                 var messagesToSend = new Queue<ServiceBusMessage>(operations.Count);
+                var messagesThatCouldntBeSent = new List<ServiceBusMessage>(operations.Count);
```

Comment:
Technically, we could also try to be smart and allocate a certain capacity here only or let the list grow if needed. That might even be better because we assume this list should be mostly empty and we are willing to pay the price of the capacity increase